src/components/ec-phone-number-input/ec-phone-number-input.vue

```diff
@@ -0,0 +1,369 @@
+<template>
+  <div
+    class="ec-phone-number-input"
+    data-test="ec-phone-number-input"
+  >
+    <label
+      v-if="label || note"
+      class="ec-phone-number-input__label"
+      :for="id"
+    >
+      <span
+        v-if="label"
+        class="ec-phone-number-input__label-text"
+        data-test="ec-phone-number-input__label-text"
+      >{{ label }}</span>
+
+      <span
+        v-if="note"
+        class="ec-phone-number-input__label-note"
+        data-test="ec-phone-number-input__label-note"
+      >{{ note }}</span>
+    </label>
+
+    <div
+      ref="popperWidthReference"
+      v-ec-tooltip="{ content: isDisabled ? disabledTooltipMessage : null }"
+      data-test="ec-phone-number-input__input-group"
+      class="ec-phone-number-input__input-group"
+    >
+      <ec-dropdown
+        :id="id"
+        v-model="countriesModel"
+        is-in-group="right"
+        data-test="ec-phone-number-input__countries"
+        class="ec-phone-number-input__countries"
+        :class="{'ec-phone-number-input__countries--is-focused': countriesHasFocus}"
+        :disabled="isDisabled"
+        :error-id="errorId"
+        :error-message="errorMessage"
+        :is-loading="areCountriesLoading"
+        :is-search-enabled="isSearchEnabled"
+        :is-sensitive="isSensitive"
+        :items="countriesItems"
+        :no-results-text="noCountriesText"
+        :placeholder="countryPlaceholder"
+        :popover-options="popoverOptions"
+        :popper-modifiers="popperModifier"
+        :search-placeholder="searchCountryPlaceholder"
+        :selected-text="selectedText"
+        @blur="countriesHasFocus = false"
+        @change="onCountryChange"
+        @focus="onFocusCountry"
+        @open="$emit('open')"
+        @after-open="$emit('after-open')"
+      >
+        <template #item="{ item }">
+          <div class="ec-phone-number-input__countries-item-wrapper tw-flex tw-items-center">
+            <span
+              class="ec-phone-number-input__countries-item-name tw-mr-4"
+              data-test="ec-phone-number-input__countries-item-name"
+            >
+              {{ item.name }}
+            </span>
+            <span
+              class="tw-text-gray-4"
+              data-test="ec-phone-number-input__countries-item-calling-code"
+            >
+              {{ item.value }}
+            </span>
+          </div>
+        </template>
+      </ec-dropdown>
+      <ec-input-field
+        v-model="numberModel"
+        class="ec-phone-number-input__number"
+        data-test="ec-phone-number-input__number"
+        is-in-group="left"
+        type="number"
```

Comment:
Maybe change this to tel, which in turn will allow all kind of text and validate on EBO side that is actually a number